pjsip/src/pjsip/sip_msg.c

```diff
@@ -885,6 +885,38 @@ static pjsip_generic_int_hdr* pjsip_generic_int_hdr_shallow_clone( pj_pool_t *po
     return hdr;
 }
 
+
+///////////////////////////////////////////////////////////////////////////////
+/*
+ * Print function for a generic array header with an arbitrary delimiter.
+ * The generic array header will use this print function with ', ' as a delimiter,
+ * the privacy header will use it with '; '.
+ */
+int pjsip_generic_array_hdr_delimited_print( pjsip_generic_array_hdr *hdr,
+                      char *buf, pj_size_t size, char *delimiter, size_t delimiter_length)
+{
+    char *p = buf, *endbuf = buf+size;
+    const pj_str_t *hname = pjsip_use_compact_form? &hdr->sname : &hdr->name;
+
+    copy_advance(p, (*hname));
+    *p++ = ':';
+    *p++ = ' ';
+
+    if (hdr->count > 0) {
+    unsigned i;
+    int printed;
+    copy_advance(p, hdr->values[0]);
+    for (i=1; i<hdr->count; ++i) {
+        copy_advance_pair(p, delimiter, delimiter_length, hdr->values[i]);
+    }
+    }
+
+    return p - buf;
```

Comment:
This function should really return a `size_t` or `ptrdiff_t`, not an integer, but that’s existing, so I guess fine.